apps/src/sites/code.org/pages/public/learn/localControlList.jsx

```diff
@@ -0,0 +1,85 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import SafeMarkdown from '@cdo/apps/templates/SafeMarkdown';
+
+export const styles = {
+  linkStyle: {
+    color: 'rgb(102, 102, 102)',
+    display: 'block',
+    padding: '5px',
+    fontSize: 'inherit',
+    textDecoration: 'none',
+    cursor: 'pointer'
+  },
+  listStyle: {
+    margin: '0px',
+    padding: '0px',
+    listStyleType: 'none'
+  }
+};
+
+export default class LocalMapControlList extends React.Component {
+  static propTypes = {
+    selected: PropTypes.number.required,
+    lng: PropTypes.string.required,
+    lat: PropTypes.string.required,
+    featureList: PropTypes.arrayOf(PropTypes.object).required,
+    flyToLocation: PropTypes.func.required,
+    createPopUp: PropTypes.func.required,
+    resetMap: PropTypes.func.required,
+    updateActive: PropTypes.func.required
+  };
+
+  render() {
+    const {
+      featureList,
+      lng,
+      lat,
+      flyToLocation,
+      createPopUp,
+      resetMap,
+      selected,
+      updateActive
+    } = this.props;
+    var allFeatures = [];
+    featureList.forEach((feature, index) => {
```

Comment:
In theory this could be another React component, even if it's just in the same file convenience.  That would make inspecting the React DOM a bit cleaner, too.

Reply:
The React DOM would particularly be cleaner if each line item was an instance of a React component.